src/renderer/vanilla/TemplateElement.js

```diff
@@ -0,0 +1,57 @@
+import { StyledElement } from '../../StyledElement';
+import { html, spreadAttributes, unsafeHTML } from './util/html';
+import { render } from './util/render';
+import { attachDeclarativeShadowDOM } from '../../util/DOMHelper';
+export { i18n } from '../../util/i18n';
+
+class TemplateElement extends StyledElement {
+	static _$templateElement$ = true;
+
+	constructor(options) {
+		super({
+			deferUpdate: false,
+			shadowRender: false,
+			styles: [],
+			adoptGlobalStyles: true,
+			mutationObserverOptions: {
+				childList: false,
+			},
+			...options,
+		});
+		this._template = this._options.template;
+
+		if (this._options.shadowRender) {
+			const internals = this.attachInternals();
```

Comment:
This here is to support existing declarative shadow dom and not attaching the shadow twice. See https://web.dev/declarative-shadow-dom/. Could make sense to port to our standard TemplateElement

Reply:
@eddyloewen @quarkus